easyrules-spring-boot-starter/src/test/java/org/easyrules/spring/boot/autoconfigure/EasyRulesAutoConfigurationTest.java

```diff
@@ -0,0 +1,161 @@
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package org.easyrules.spring.boot.autoconfigure;
+
+import org.easyrules.annotation.Action;
+import org.easyrules.annotation.Condition;
+import org.easyrules.annotation.Rule;
+import org.easyrules.api.RulesEngine;
+import org.easyrules.core.BasicRule;
+import org.easyrules.spring.RulesEngineFactoryBean;
+import org.easyrules.spring.SpringRule;
+import org.junit.After;
+import org.junit.Test;
+import org.springframework.boot.test.util.EnvironmentTestUtils;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.stereotype.Component;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Tests for {@link EasyRulesAutoConfiguration}
+ *
+ * @author Kyle Anderson
+ */
+public class EasyRulesAutoConfigurationTest {
+
+    private AnnotationConfigApplicationContext context;
+
+    @After
+    public void tearDown() {
+        if(this.context != null) {
+            this.context.close();
+        }
+    }
+
+    @Test
+    public void rulesEngineFactoryBeanTest() {
+        load(EmptyConfiguration.class);
+        RulesEngineFactoryBean rulesEngineFactoryBean = this.context.getBean(RulesEngineFactoryBean.class);
+        assertNotNull(rulesEngineFactoryBean);
+    }
+
+    @Test
+    public void rulesEngineBeanTest() {
+        load(EmptyConfiguration.class);
+        RulesEngine rulesEngine = this.context.getBean(RulesEngine.class);
+        assertNotNull(rulesEngine);
+    }
+
+    @Test
+    public void noRulesConfigured() {
+        load(EmptyConfiguration.class);
+        RulesEngine rulesEngine = this.context.getBean(RulesEngine.class);
+        assertEquals(0, rulesEngine.getRules().size());
+    }
+
+    @Test
+    public void ruleImplementationBeanTest() {
+        load(RuleImplementationBeanConfiguration.class);
+        RulesEngine rulesEngine = this.context.getBean(RulesEngine.class);
+        assertEquals(1, rulesEngine.getRules().size());
+    }
+
+    @Test
+    public void ruleComponentAnnotationTest() {
+        load(RuleComponentAnnotationConfiguration.class);
+        RulesEngine rulesEngine = this.context.getBean(RulesEngine.class);
+        assertEquals(1, rulesEngine.getRules().size());
+    }
+
+    @Test
+    public void springRuleAnnotationTest() {
+        load(SpringRuleAnnotationConfiguration.class);
+        RulesEngine rulesEngine = this.context.getBean(RulesEngine.class);
+        assertEquals(1, rulesEngine.getRules().size());
+    }
+
```

Comment:
I think we can add a test case when we have multiple configuration classes:

```java
@Test
public void multipleRulesTest() {
    load(SpringRuleAnnotationConfiguration.class);
    load(RuleComponentAnnotationConfiguration.class);
    load(RuleComponentAnnotationConfiguration.class);
    // TODO adapt the load method to take multiple configuration classes
    RulesEngine rulesEngine = this.context.getBean(RulesEngine.class);
    assertEquals(3, rulesEngine.getRules().size());
}
```

What do you think?